src/GraphQL/Utilities/SchemaPrinter.cs

```diff
@@ -338,28 +347,37 @@ private string FormatInputObjectValue(object value, IInputObjectGraphType input)
             var sb = new StringBuilder();
             sb.Append("{ ");
 
-            foreach (var field in input.Fields)
+            foreach (var field in input.Fields.OrderBy(Options.Comparer?.FieldComparer(input)))
             {
                 string propertyName = field.GetMetadata<string>(ComplexGraphType<object>.ORIGINAL_EXPRESSION_PROPERTY_NAME) ?? field.Name;
-                PropertyInfo propertyInfo;
 
-                try
+                // if 'value' is stored as a dictionary of key/value pairs, pull the field value from the dictionary by the property name
+                object? propertyValue;
+                if (value is IDictionary<string, object?> dic)
                 {
-                    propertyInfo = value.GetType().GetProperty(propertyName, BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance);
+                    // note: per spec, unspecified properties may not exist within the dictionary
+                    if (!dic.TryGetValue(propertyName, out propertyValue)) continue;
                 }
-                catch (AmbiguousMatchException)
+                // if 'value' is stored as an object -- e.g. new MyObject { Value = "Test" } -- then pull from the property directly
+                else
                 {
-                    propertyInfo = value.GetType().GetProperty(propertyName, BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+                    PropertyInfo propertyInfo;
+                    try
+                    {
+                        propertyInfo = value.GetType().GetProperty(propertyName, BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance);
+                    }
+                    catch (AmbiguousMatchException)
+                    {
+                        propertyInfo = value.GetType().GetProperty(propertyName, BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+                    }
+
+                    propertyValue = propertyInfo.GetValue(value);
                 }
 
-                object propertyValue = propertyInfo.GetValue(value);
-                if (propertyValue != null)
```

Comment:
Skipping null members can produce invalid schema, as seen by the added test.

Reply:
👍